src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -225,15 +237,28 @@ else if (editMode == 'C') {
                 }
             }
         }
-
         this.comment = message.toString();
-
-        int endOfFirstLine = this.comment.indexOf('\n');
-        if (endOfFirstLine == -1) {
-            this.title = this.comment;
+        if (!truncateMessageTitle) {
```

Comment:
Isn't the sense of this boolean backwards?

When `truncateMessageTitle` is true, I expected that splitString would be called with TRUNCATE_LIMIT.  Instead, when `truncateMessageTitle` is false, it calls splitString with TRUNCATE_LIMIT.

Shouldn't the variable be renamed to `retainFullCommitSummary` or something like that?

Reply:
I think using *truncate* was not a good idea, because is always truncated. The method _splitString_ just gets whatever is first: new line or _TRUNCATE_LIMIT_.
 Having said that, I think you suggestion is great. I will the changes